lib/spack/docs/packaging_guide.rst

```diff
@@ -4054,36 +4054,193 @@ File functions
 Making a package discoverable with ``spack external find``
 ----------------------------------------------------------
 
-To make a package discoverable with
-:ref:`spack external find <cmd-spack-external-find>` you must
-define one or more executables associated with the package and must
-implement a method to generate a Spec when given an executable.
+The simplest way to make a package discoverable with
+:ref:`spack external find <cmd-spack-external-find>` is to:
 
-The executables are specified as a package level ``executables``
-attribute which is a list of strings (see example below); each string
-is treated as a regular expression (e.g. 'gcc' would match 'gcc', 'gcc-8.3',
-'my-weird-gcc', etc.).
+1. Mark the Package with the ``detectable`` decorator
+2. Define the executables associated with the package
+3. Implement a method to determine the versions of these executables
 
-The method ``determine_spec_details`` has the following signature:
+^^^^^^^^^^^^^^^^^
+Minimal detection
+^^^^^^^^^^^^^^^^^
+
+The first two steps are fairly simple, as they require only to
+decorate the package being defined and specify a package level
+``executables`` attribute:
+
+.. code-block:: python
+
+   @detectable
+   class Foo(Package):
+       # Each string provided here is treated as a regular expression, and
+       # would match for example 'foo', 'foobar', and 'bazfoo'.
+       executables = ['foo']
+
+This attribute must be a list of strings. Each string is treated as a regular
+expression (e.g. 'gcc' would match 'gcc', 'gcc-8.3', 'my-weird-gcc', etc.) to
+determine a set of system executables that might be part or this package. Note
+that to match only executables named 'gcc' the regular expression ``r'^gcc$'``
+must be used.
+
+Finally to determine the version of each executable the ``determine_version``
+method must be implemented:
+
+.. code-block:: python
+
+   @classmethod
+   def determine_version(cls, exe):
+       """Return either the version of the executable passed as argument
+       or ``None`` if the version cannot be determined
+       """
+
+This method receives as input the path to a single executable and must return
+as output its version as a string, or ``None`` is the version cannot be determined.
```

Comment:
> One nice feature of the original implementation of determine_spec_details was that if it returned None, it was assumed that the provided executable wasn’t an instance of the package, does determine_version have this property?

Yes, it has the same property (in details if the version is false-ish the executable won't be added to `exes_by_version` and thus won't proceed in the workflow), see:

https://github.com/spack/spack/blob/b52c236a2ca7843042e6049c6e06596aacb2b9aa/lib/spack/spack/package.py#L2238-L2246

> Perhaps the same outcome could be achieved by raising a special-purpose exception if determine_version realizes that the exe isn't actually an instance of the package.: "NotAnInstanceOfPackage".

Not sure this would be an improvement or a complication for users. In general with the current version users just need to return a value. With the modification proposed they need to return a value or raise an appropriate exception. For the time being I would keep the semantics as it is and maybe modify it in a later PR when we have more experience on a broader set of use cases.

Reply:
> Yes, it has the same property (in details if the version is false-ish the executable won't be added to exes_by_version 

Excellent: can you mention that in the docs here?

Reply:
Done